ale/teacher/exploitation/tag_flip_historical.py

```diff
@@ -55,37 +71,36 @@ def __init__(self, docs_with_predictions: Dict[int, PredictionResult]) -> None:
         Helper class used to wrap predictions of last cycle.
 
         Args:
-            - docs_with_predictions (Dict[int,PredictionResult]): Dict containing the ids of the documents and their prediction result
+            - docs_with_predictions (Dict[int,PredictionResult]): Dict containing the ids of the documents
+                                                                  and their prediction result
         """
-        self.docs: List[HistoricalDoc] = []
-        for doc in docs_with_predictions:
-            historical_doc: HistoricalDoc = HistoricalDoc(
-                doc, docs_with_predictions[doc])
-            self.docs.append(historical_doc)
-
-    def update_and_compare_historical_sequences(self, docs_with_predictions: Dict[int, PredictionResult]) -> Dict[int, List[int]]:
+        self.docs: Dict[int, HistoricalDoc] = {}
+        for doc_idx, predictions in docs_with_predictions.items():
+            historical_doc: HistoricalDoc = HistoricalDoc(doc_idx, predictions)
+            self.docs[doc_idx] = historical_doc
+
+    def update_and_compare_historical_sequences(self,
+                                                docs_with_predictions: Dict[int, PredictionResult]) -> Dict[
+        int, List[int]]:
         differences: Dict[int, List[int]] = {}
-        for doc in docs_with_predictions:
-            historical_doc: Optional[HistoricalDoc] = self.get_historical_doc_by_id(
-                doc)
-            if historical_doc:  # history exists for document
-                difference: List[int] = historical_doc.update_and_compare_labels(
-                    docs_with_predictions[doc])
-                differences[doc] = difference
-            else:
-                historical_doc: HistoricalDoc = HistoricalDoc(
-                    doc, docs_with_predictions[doc])
-                # doc never predicted before, so no difference
-                difference: List[int] = [
-                    0 for item in historical_doc.get_labels()]
-                differences[doc] = difference
+        for doc_idx, prediction in docs_with_predictions.items():
+            differences[doc_idx] = self.compute_difference_for_doc(doc_idx, prediction)
         return differences
 
+    def compute_difference_for_doc(self, doc_idx: int, prediction: PredictionResult):
```

Comment:
Finde den Namen der Methode ein bisschen irreführend. Vielleicht update_and_compute_difference_for_doc? Oder update_history einzeln und dann compute_diffs?

Reply:
Good point! I changed the method to update only the historical_doc and return it. Then, compute the differences in the for loop. Please take a look at it again.

Reply:
Yes, looks good now